sstables/sstables.hh

```diff
@@ -960,3 +964,36 @@ future<> remove_table_directory_if_has_no_snapshots(fs::path table_dir);
 future<> remove_by_toc_name(sstring sstable_toc_name);
 
 } // namespace sstables
+
+template <>
+struct fmt::formatter<sstables::sstable> {
+    std::optional<sstables::component_type> f = std::nullopt;
+
+    constexpr auto parse(format_parse_context& ctx) {
```

Comment:
have you considered keeping the implementation of formatter in `sstables.cc` ? see #14020

Reply:
I have, but I have some irrational strong "against" instantiating code-wide templates in .cc files :( Maybe I'll try it in v2

Reply:
I agree the split isn't nice. If we could avoid all of the specialization in the header file, it would be worthwhile, but we only avoid it partially, so the code gets quite ugly.